google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -236,6 +196,33 @@ private void installAppEngineJavaComponent()
     }
   }
 
+  private Path updateManagedSdk() throws Exception {
+    if (!safeCheckSdkStatus(() -> managedCloudSdk.isUpToDate())) {
+      MessageListener sdkUpdateListener = logger::debug;
+
+      managedCloudSdk.newUpdater().update(sdkUpdateListener);
+    }
+
+    return managedCloudSdk.getSdkHome();
+  }
+
+  /**
+   * Checks SDK status (installed, up-to-date, has component), catching non-fatal exceptions which
+   * does not mean SDK cannot be updated or re-installed. Other exceptions are propagated to the
+   * caller.
```

Comment:
Is there example here of these exceptions being propagated to and caught by the caller? or is that in the next PRs?

Reply:
There's a few examples in the unit tests (IOException etc.). Do you mean we might add them into javadoc?

Reply:
I didn't mean add them to the javadoc. I just wanted get an idea of how we going to handle these exceptions that are not caught (in non-test code).

Reply:
As of now all error handling is in `ManagedSdkJobListener.onFailure()`

Reply:
oh I missed that. looks good